### PR TITLE
feat: set all fields on success x-origin auth

### PR DESCRIPTION
**Pull request by danielgent**

### CODE REVIEW

File: src/authentication-flows/ticket.ts [hunk 1]
```diff
@@ -86,6 +86,7 @@ export async function ticketAuth(
   }
 
   ctx.set("userId", user.id);
+  ctx.set("userName", user.name || user.email);
```
Comment (danielgent) [inline src/authentication-flows/ticket.ts:89]: Seems to be how Auth0 does it... `user_name` can be name or email :thinking:

File: src/authentication-flows/social.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ import { getClient } from "../services/clients";
 import { LogTypes } from "../types";
 
 export async function socialAuth(
-  env: Env,
+  ctx: Context<{ Bindings: Env; Variables: Var }>,
```
Comment (danielgent) [inline src/authentication-flows/social.ts:23]: as mentioned on a previous PR, we need this everywhere to set the log type

File: src/tsoa-middlewares/logger.ts [hunk 3]
```diff
@@ -58,6 +58,13 @@ function createTypeLog(
       };
       return successApiOperation;
     case "scoa":
+      // if (!ctx.var.userId)
+      //   throw new Error("userId is required for scoa log type");
+      // if (!ctx.var.userName)
+      //   throw new Error("userName is required for scoa log type");
+      // if (!ctx.var.connection)
+      //   throw new Error("connection is required for scoa log type");
```
Comment (danielgent) [inline src/tsoa-middlewares/logger.ts:66]: helpers for debugging. I can remove these before each PR and put them back in, else I'll remove them at the end

File: src/routes/tsoa/callback.ts [hunk 4]
```diff
@@ -114,6 +114,16 @@ export class CallbackController extends Controller {
...
 
+    request.ctx.set("client_id", loginState.authParams.client_id);
+    const client = await getClient(
+      request.ctx.env,
+      loginState.authParams.client_id,
+    );
+    if (!client) {
+      throw new HTTPException(400, { message: "Client not found" });
+    }
+    request.ctx.set("tenantId", client.tenant_id);
```
Comment (markusahlstrand) [inline src/routes/tsoa/callback.ts:125]: Are we doing camel casing here? Should we snake case it as well?

File: src/tsoa-middlewares/logger.ts [hunk 5]
```diff
@@ -182,6 +193,9 @@ export function loggerMiddleware(
 
       const logType = ctx.var.logType || logTypeInitial;
 
+      if (DEBUG_LOG_TYPES && !ctx.var.tenantId)
```
Comment (markusahlstrand) [inline src/tsoa-middlewares/logger.ts:196]: I like the curly braces :)

File: src/types/Var.ts [hunk 6]
```diff
@@ -15,12 +15,13 @@ export type Var = {
     permissions: string[];
   };
   userName?: string;
+  // I'm not sure what connection_id actually means here...  In Auth0 we get con_TI7p6dEHf551Q9t6
+  // connectionId?: string;
```
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply src/types/Var.ts:19]: No idea what it is..

**Review by markusahlstrand — APPROVED** [review]
Looks good

File: src/routes/tsoa/callback.ts (hunk 4 above)
Comment (danielgent) [reply src/routes/tsoa/callback.ts:125]: I just continued on with what you already had here! We can always switch these casing in the future they're easy enough to change with typescript

File: src/tsoa-middlewares/logger.ts (hunk 5 above)
Comment (danielgent) [reply src/tsoa-middlewares/logger.ts:196]: oh yeah. I try and put this kind of thing on one line but it must be prettier.

We can set this as a lint rule. I thought it was by default.  It was in Crockford's JS lint anyway

**Pull request merged**